### ISSUE TITLE
Clarify if and errexit paragraph

### ISSUE THREAD

**Issue by happysalada** [oilshell/oil#934]
In the idioms https://www.oilshell.org/release/latest/doc/idioms.html#if-and-errexit
the rule around if and brackets is not completely clear.
I would add at the end of that paragraph, something like
```
note that you can't add brackets around a function.
brackets are for evaluating conditionals only.

    if (! myfunc) {
        ^
   Syntax error in expression (near Id.Expr_Bang)
```

Let me know if that makes sense. 
The original error I came from was this

```
./run.sh dev
    if (! pg_isready) {
        ^
./run.sh:4: Syntax error in expression (near Id.Expr_Bang)
```

if that works for you, I'll make the PR, just let me know.

**Comment by andychu** [comment]
I think a one-sentence link to "A Tour of Oil" that reminds the user that parens are significant would be very helpful!  I want to keep this doc brief, but it's worth mentioning.

**Comment by andychu** [comment]
It's also sorta mentioned here: https://www.oilshell.org/release/0.8.10/doc/syntax-feelings.html#parens-expression

But I agree that the significant parens are unlike other languages, so it's worth highlighting

**Comment by andychu** [comment]
Reopening this since we still haven't published a "Tour of Oil" yet !

I want to also thank you for the great feedback and say I didn't forget about this!  We have `build/doc.sh tour` which now actually validates the code in the doc.  And the next step is to actually finish writing it!

I have been travelling, so that has been unfortunately delayed, but this is one of the more important things coming up!

**Comment by happysalada** [comment]
When you have a moment, could you detail the bigger parts that are left to write?
If you give me just a 1-10 bullet point list, I can dig into each of those and take a stab at it.
Like usual, If whatever I write doesn't end up being the actual thing but just serves as inspiration, I'm happy.

**Comment by andychu** [comment]
Hm I think it would be useful to add more to the `oil_lang/testdata/tour.oil` file?  And to make sure it runs with `bin/oil`.  Remember there was an outline in the PR.

I copied some of that into the draft of `doc/tour.md`, so it does help!

**Comment by happysalada** [comment]
Just to make sure I understand correctly.
Here is the original outline


    command language
        simple commands
            builtins
            external commands
            Ruby-like blocks -- e.g. cd /tmp { pwd }. This still needs more implementation.
        redirects
        pipelines
        control flow
            if
            case
            for
            while
        procs
        variable declaration and mutation
        maybe: fork and wait
    word language
        variable sub - $var
        command sub - $(hostname)
        expression sub - $[1 + 2 * 3]
        splicing @myarray
        inline function calls like @split(x) and $join()
    expression language
        this is mostly Python
        eggex


Here is what I understand is still missing

```
### external commands

### Ruby like blocks

## Redirects

## Pipelines

### case

### for

### while

## fork and wait
```
(this first part was directly taken from the missing stuff in tour.oil)

And those as well (taken from the previous outline)
```
    expression sub - $[1 + 2 * 3]
        splicing @myarray
        inline function calls like @split(x) and $join()
```

Let me know if you had something else in mind or think some of the things that were already treated should be improved.

**Comment by andychu** [comment]
Yes that's what I was referring to.  One way you can check it over is to show it to a friend or coworker and see if it makes sense!  (I think we're still very far from that -- docs are a lot of work)

**Comment by andychu** [comment]
BTW on the plane ride yesterday I made a little progress on this:

https://github.com/oilshell/oil/blob/master/doc/oil-language-tour.md (ROUGH)

I used the outline, but then of course writing it made me realize that I want to re-order it: into word, command, and expression languages.  Those are really the 3 core components of the Oil language (and I had to write it to figure it out!)

The `if` issue is in there, but needs to be clearer.

I also enhanced the toolchain a bit, and found a BUG in the language!  That is one reason to write the docs.

-----

I think it's still helpful if you write a draft in the comments, because I can use some of that material, see a different perspective, and I won't be surprised if you find bugs as well!  Different people will tickle the language in different ways.

But if you don't want to step on each other's toes, there are a lot of other issues tagged #documentation which perhaps can use help.

I was traveling for 3 weeks, but this is one of my top priorities, as mentioned on the blog:

https://www.oilshell.org/blog/2021/06/oil-language.html#whats-next
